src/backtick/core.clj

```diff
@@ -15,35 +20,56 @@
          f name)
   name)
 
-(defn unregister [name]
+(defn unregister
+  "Unregisters a Backtick worker."
+  [name]
   (swap! engine/workers dissoc name (@engine/workers name))
   name)
 
-(defn register-cron
-  "Register or re-register a scheduled job.  The job is started for the
-   first time interval ms from now."
-  [name interval f]
-  (register name f)
-  (engine/cron-add name interval)
-  name)
-
-(defn registered-workers []
-  @engine/workers)
+(defn registered-workers
+  "Returns a map of all registered workers, keyed by name."
+  []
+  (->> @engine/workers
+       (filter #(string? (key %)))
+       (into {})))
 
-(defn registered-crons
-  "All the cron jobs as a map"
+(defn recurring-jobs
+  "Returns a map of all recurring jobs, keyed by name."
   []
   (engine/cron-map))
 
+(defn- resolve-worker [worker]
+  (let [v (@engine/workers worker)]
+    (cond
+      (nil? v) (do
+                 (log/errorf (str "Worker %s not registered. Call register on this "
+                                  "function before attempting to schedule it.")
+                                  worker)
+                 nil)
+      (string? v) v
+      :else worker)))
```

Comment:
It seems like `engine` wants a function reference in its `add` functions. Since the workers map is 2-way, this ensures that our `schedule*` functions can take _either_ the worker name or a function reference and abstract over that distinction.